contrib/win32/win32compat/w32fd.c

```diff
@@ -812,19 +812,22 @@ int w32_allocate_fd_for_handle(HANDLE h, BOOL is_sock) {
 
 int 
 w32_ftruncate(int fd, off_t length) {
-    CHECK_FD(fd);
+	LARGE_INTEGER new_postion;
+
+	CHECK_FD(fd);
 
-    if (!SetFilePointer(w32_fd_to_handle(fd), length, 0, FILE_BEGIN))
-        return -1;
-    if (!SetEndOfFile(w32_fd_to_handle(fd)))
-        return -1;
+	new_postion.QuadPart = length;
+	if (!SetFilePointerEx(w32_fd_to_handle(fd), new_postion, 0, FILE_BEGIN))
```

Comment:
curious to understand why it is working for sftp.

also check the sftp put/get with the new changes.